frontend/src/metabase/common/components/FilterPicker/NumberFilterPicker.tsx

```diff
@@ -1,31 +1,215 @@
 import { t } from "ttag";
-import { Box, Button, Flex, Text } from "metabase/ui";
+import { useState, useMemo } from "react";
+import { Box, Button, Flex, Text, NumberInput } from "metabase/ui";
 import * as Lib from "metabase-lib";
+
+import FieldValuesWidget from "metabase/components/FieldValuesWidget";
+import Field from "metabase-lib/metadata/Field";
 import type { FilterPickerWidgetProps } from "./types";
 import { BackButton } from "./BackButton";
 import { Header } from "./Header";
 import { Footer } from "./Footer";
 
+import { FilterOperatorPicker } from "./FilterOperatorPicker";
+
+const numberValueTypes = ["single", "multiple", "two", "none"];
+type NumberValueType = typeof numberValueTypes[number];
+
+const operatorTypes: Record<NumberValueType, Lib.FilterOperatorName[]> = {
+  single: [">", ">=", "<", "<="],
+  multiple: ["=", "!="],
+  two: ["between"],
+  none: ["is-null", "not-null", "is-empty", "not-empty"],
+};
```

Comment:
I don't love this

Reply:
This might be improved but the overall approach would be very close. I'm thinking about something like this:
```
type OptionArguments = 0 | 1 | 2 | "multiple";

interface Option {
  operator: Lib.FilterOperator;
  shortName: Lib.FilterOperatorName;
  displayName: string;
  arguments: OptionArguments;
}

function getOptions(query: Lib.Query, stageIndex: number, column: Lib.ColumnMetadata): Option[] {
  const operators = Lib.filterableColumnOperators(column);
  return operators.reduce((options: Option[], operator) => {
    const operatorInfo = Lib.displayInfo(query, stageIndex, operator);
    switch(operatorInfo.shortName) {
      case "is-null":
      case "not-null":
      case "is-empty":
      case "not-empty":
        options.push({ operator, ...operatorInfo, arguments: 0 });
        break;
      case ">":
      case ">":
        ...
        options.push({ operator, ...operatorInfo, arguments: 1 });
    };
  }, []);
}
```

Reply:
This would be close to `BooleanFilterPicker`

Reply:
@ranquild pardon, I edited your message to add syntax highlighting to the snippet

Reply:
I like that suggestion and the fact that we're keeping filter components similar — makes it easier to switch from one to another. My only minor-ish comment is to use a name like `argsCount` or whatever: `arguments` doesn't make it clear  it's about the expected number of args